rasa/nlu/data_augmentation/augmenter.py

```diff
@@ -0,0 +1,736 @@
+    significant_figures: int = 2,
+) -> Set[Text]:
+    """Extracts the intents whose performance has changed.
+
+    Args:
+        classification_report_no_augmentation: Classification report of the model run
+            *without* data augmentation.
+        intent_report_with_augmentation: Report of the model run
+            *with* data augmentation.
+        all_intents: List of all intents.
+        significant_figures: Significant figures to be taken into account when
+            assessing whether the performance of an intent has changed.
+
+    Returns:
+        Set of intents that have changed - i.e. that were affected by data augmentation.
+    """
+    changed_intents = set()
+    for intent_key in all_intents:
+        for metric in ["precision", "recall", "f1-score"]:
+            rounded_original = round(
+                classification_report_no_augmentation.get(intent_key, {}).get(
+                    metric, 0.0
+                ),
+                significant_figures,
+            )
+            rounded_augmented = round(
+                intent_report_with_augmentation.get(intent_key, {}).get(metric, 0.0),
+                significant_figures,
+            )
+            if rounded_original != rounded_augmented:
+                changed_intents.add(intent_key)
+
+    return changed_intents
+
+
+def _run_data_augmentation(
+    nlu_training_data: TrainingData,
+    intents_to_augment: Set[Text],
+    nlu_training_file: Text,
+    output_directory: Text,
+    config: Text,
+    nlu_evaluation_data: TrainingData,
+    classification_report_no_augmentation: Dict[Text, Dict[Text, float]],
+) -> None:
+    """Runs the NLU train/test cycle with data augmentation.
+
+    Also, generate reports and plots summarising the impact of data augmentation
+    on model performance.
+
+    Args:
+        nlu_training_data: The augmented NLU training data.
+        intents_to_augment: The intents that were chosen for data augmentation.
+        nlu_training_file: The file for storing the augmented NLU training data.
+        output_directory: Directory to store the output files in.
+        config: NLU model config.
+        nlu_evaluation_data: NLU evaluation data.
+        classification_report_no_augmentation: Classification report of the model run
+            *without* data augmentation.
+    """
+    # Store augmented training data to file
+    nlu_training_data.persist_nlu(filename=nlu_training_file)
+
+    # Run NLU train/test loop
+    intent_report = _train_test_nlu_model(
+        output_directory=output_directory,
+        nlu_training_file=nlu_training_file,
+        config=config,
+        nlu_evaluation_data=nlu_evaluation_data,
+    )
+
+    # Create data augmentation summary
+    intent_summary = _create_summary_report(
+        intent_report_with_augmentation=intent_report,
+        classification_report_no_augmentation=classification_report_no_augmentation,
+        training_intents=nlu_training_data.intents,
```

Comment:
pycharm says you're passing incompatible types?

Reply:
dunno, could have been `Dict[Text, Any] vs. Dict[Text, Dict[Text, float]]`